pkg/docker/detection.go

```diff
@@ -0,0 +1,9 @@
+package docker
+
+import "os"
+
+// IsRunningInDocker checks whether the current process is being executed inside a docker container
```

Comment:
Does it work with other runtimes (e.g. podman, crio)?
e.g. 
- https://sourcegraph.com/github.com/composer/composer/-/blob/src/Composer/Util/Platform.php?L199-201
- https://sourcegraph.com/github.com/FunnyWolf/vipermsf/-/blob/lib/msf/core/post/linux/system.rb?L405-436

Reply:
I've extended detection to check for the other files that your examples use to check for docker, and I've also looked up what to check for regarding cryo and podman.

Any others we should check for? I'm not that familiar with the landscape.

Reply:
In theory we can check for handcrafted cgroups but I think it's not popular approach :)

Reply:
I'll leave those out then, it can be added later if needed